mvpa2/algorithms/searchlight_hyperalignment.py

```diff
@@ -0,0 +1,488 @@
+            os.unlink(results)
+            if __debug__:
+                debug('SLC_', "Loaded results of len=%d from"
+                      % len(results_data))
+            for isub, res in enumerate(results_data):
+                self.projections[isub] = self.projections[isub] + res
+                if self.params.compute_recon:
+                    self.projections_recon[isub] = self.projections_recon[isub] + res.T
+            return
+
+    def __handle_all_results(self, results):
+        """Helper generator to decorate passing the results out to
+        results_fx
+        """
+        for r in results:
+            yield self.__handle_results(r)
+
+    def __call__(self, datasets):
+        """Estimate mappers for each dataset using searchlight-based
+        hyperalignment.
+
+        Parameters
+        ----------
+          datasets : list or tuple of datasets
+
+        Returns
+        -------
+        A list of trained StaticProjectionMappers of the same length as datasets
+        """
+        self.ndatasets = len(datasets)
+        params = self.params
+        _shpaldebug("SearchlightHyperalignment %s for %i datasets" % (self, self.ndatasets))
+        if params.ref_ds in params.exclude_from_model:
+            raise (ValueError, "Requested reference dataset %i is also "
+                               "in the exclude list." % params.ref_ds)
+        if params.ref_ds != params.hyperalignment.params.ref_ds:
+            warning('Supplied ref_ds & hyperalignment instance ref_ds:%d differ.'
+                    % params.hyperalignment.params.ref_ds)
+            warning('Using default hyperalignment instance with ref_ds: %d' % params.ref_ds)
+            params.hyperalignment = Hyperalignment(ref_ds=params.ref_ds)
+        if params.ref_ds >= self.ndatasets:
+            raise (ValueError, "Requested reference dataset %i is out of "
+                               "bounds. We have only %i datasets provided" % (params.ref_ds, self.ndatasets))
+        if len(params.exclude_from_model) > 0:
+            warning("These datasets will not participate in building common model: %s" % params.exclude_from_model)
+        # Setting up SearchlightHyperalignment
+        # we need to know which original features where comprising the
+        # individual SL ROIs
+        _shpaldebug('Initializing HyperalignmentMeasure.')
+        hmeasure = HyperalignmentMeasure(featsel=params.featsel,
+                                         hyperalignment=params.hyperalignment,
+                                         full_matrix=params.combine_neighbormappers,
+                                         use_same_features=params.use_same_features,
+                                         exclude_from_model=params.exclude_from_model, dtype=params.dtype)
+        # Performing SL processing manually
+        _shpaldebug("Setting up for searchlights")
+        if params.nproc is None and externals.exists('pprocess'):
```

Comment:
FYI better to not override params.nproc  but rather assign local nproc variable and manipulate that one

anyways -- I will RF this one to share functionality with searchlight which has similar check/etc